### PR TITLE
Add feature to allow empty root password

### PR DESCRIPTION
**Pull request by Hoega**
Close https://github.com/mariadb-operator/mariadb-operator/issues/323

This PR introduces the rootEmptyPassword option to allow a simple mariadb connection without a password.

```
apiVersion: mariadb.mmontes.io/v1alpha1
kind: MariaDB
metadata:
  name: mariadb
spec:
  rootEmptyPassword: true
```

If rootEmptyPassword = true, the RootPasswordSecretKeyRef option will be ignored and the root password will be empty.
When rootEmptyPassword = false, the default behavior is unchanged: a random root password will be generated if a secret is not already provided.

With rootEmptyPassword = true, I can connect to the database as root without a password:
```
k -it exec mariadb-0 -- mariadb -D mariadb  -u root
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 152
Server version: 11.0.3-MariaDB-1:11.0.3+maria~ubu2204 mariadb.org binary distribution

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [mariadb]> CREATE TABLE t1 ( a INT );
Query OK, 0 rows affected (0.025 sec)

```

Whereas if the rootEmptyPassword = false or nil, a password is required:
```
k -it exec mariadb-0 -- mariadb  -D mariadb  -u root
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: NO)
command terminated with exit code 1
```

### CODE REVIEW

File: api/v1alpha1/mariadb_types.go [hunk 1]
```diff
@@ -345,9 +349,15 @@ func (m *MariaDB) SetDefaults(env *environment.Environment) {
 	if m.Spec.Image == "" {
 		m.Spec.Image = env.RelatedMariadbImage
 	}
-	if m.Spec.RootPasswordSecretKeyRef == (corev1.SecretKeySelector{}) {
+
+	if m.Spec.RootEmptyPassword == nil {
+		defaultRootEmptyPassword := false
+		m.Spec.RootEmptyPassword = &defaultRootEmptyPassword
```
Comment (mmontes11) [inline api/v1alpha1/mariadb_types.go:355]: The [k8s.io/utils/ptr](https://pkg.go.dev/k8s.io/utils/ptr) pkg is a very nice utility I am trying to migrate to progresively, can we use it here?
```go
ptr.To(false)
```
See:
https://github.com/mariadb-operator/mariadb-operator/blob/d2ee5de461323cd5eb29b22b8952a39b0cc5ebb0/api/v1alpha1/mariadb_types_test.go#L82

File: api/v1alpha1/mariadb_types.go [hunk 2]
```diff
@@ -408,6 +418,11 @@ func (m *MariaDB) IsInitialDataEnabled() bool {
 	return m.Spec.Username != nil
 }
 
+// IsRootPasswordEmpty indicates whether the MariaDB instance has an empty root password
+func (m *MariaDB) IsRootPasswordEmpty() bool {
```
Comment (mmontes11) [inline api/v1alpha1/mariadb_types.go:422]: Great 👍🏻

File: pkg/builder/statefulset_container_builder.go [hunk 3]
```diff
@@ -345,7 +354,7 @@ var (
 				Command: []string{
 					"bash",
 					"-c",
-					"mariadb -u root -p\"${MARIADB_ROOT_PASSWORD}\" -e \"SHOW STATUS LIKE 'wsrep_ready'\" | grep -c ON",
+					"mariadb -u root -p\"$MARIADB_ROOT_PASSWORD\" -e \"SHOW STATUS LIKE 'wsrep_ready'\" | grep -c ON; ",
```
Comment (mmontes11) [inline pkg/builder/statefulset_container_builder.go:357]: Why removing the curly braces here?

**Review by mmontes11 — COMMENTED** [review]
Looking good so far, left some comments

**Comment by mmontes11** [comment]
CI is complaining about not having generated artifacts:
- https://github.com/mariadb-operator/mariadb-operator/actions/runs/7431141447/job/20221581492?pr=328

Just run `make gen` and commit. This is to make sure the CRDs, documentation and all the generated stuff is up to date.

File: api/v1alpha1/mariadb_types.go (hunk 1 above)
Comment (Hoega) [reply api/v1alpha1/mariadb_types.go]: Done

File: pkg/builder/statefulset_container_builder.go (hunk 3 above)
Comment (Hoega) [reply pkg/builder/statefulset_container_builder.go]: It was by mistake, I reverted

**Comment by Hoega** [comment]
> CI is complaining about not having generated the artifacts:
> 
> * https://github.com/mariadb-operator/mariadb-operator/actions/runs/7431141447/job/20221581492?pr=328
> 
> Just run `make gen` and commit. This is to make sure the CRDs, documentation and all the generated stuff are up to date. This includes our brand new [API reference](https://github.com/mariadb-operator/mariadb-operator/blob/main/docs/API_REFERENCE.md) automatically generated from CRDs which I've added just yesterday, so be sure to rebase with `main`.

I rebased and generated the API documentation.

**Review by mmontes11 — APPROVED** [review]
LGTM! Thanks a lot for your contribution! 🚀 🦭 

FYI @outductor

**Pull request merged**